storage/grpc_client.go

```diff
@@ -592,6 +599,16 @@ func (c *grpcStorageClient) UpdateObject(ctx context.Context, params *updateObje
 
 	req.UpdateMask = fieldMask
 
+	if len(fieldMask.Paths) < 1 {
+		// Nothing to update. To maintain consistency with JSON, we must still
+		// update the object because metageneration and other fields are
+		// updated even on an empty update.
+		// gRPC will fail if the fieldmask is empty, so instead we add an
+		// output-only field to the update mask. Output-only fields are (and must
+		// be - see AIP 161) ignored, but allow us to send an empty update.
```

Comment:
We should maybe clarify that this is intended behavior. Feels inconsistent that empty updates are barred but this is permitted.

Reply:
Added the reasoning behind why this is permitted to the comment.